packages/cosmo-dycore/package.py

```diff
@@ -63,9 +63,10 @@ class CosmoDycore(CMakePackage):
     depends_on('gridtools@1.1.3 +cuda', when='+cuda+gt1')
     depends_on('boost@1.67.0')
     depends_on('serialbox@2.6.0', when='+build_tests')
-    depends_on('mpi', type=('build', 'run'))
+    depends_on('mpicuda', type=('build', 'run'), when='+cuda')
+    depends_on('mpi', type=('build', 'run'), when='~cuda')
```

Comment:
Shouldn't mpi also be `'link'`?

Reply:
link dependencies just mean spack is appending them to the LIBRARY_PATH. Don't know if we actually need the mpilib to run cosmo ?